### PR TITLE
Vnet_route_check Vxlan tunnel route update.

### PR DESCRIPTION
**Pull request by siqbal1986**
Modified Vnet_route_check to ignore the Vxlan tunnel routes which uses default VRF.

#### What I did
Updated the vnet_route_check script to handle the vxlan tunnel routes.
#### How I did it
By checking the scope of the tunnel route, if it is default then it is treated in a different way. Tunnel routes coming from App_db are compared with ASIC_db default_VRF routes and their presence is verified.
For routes coming from ASIC_db with default_VRF, the comparison is not done with the routes form App_db as ASIC_DB carries all other the routes.
For normal vnet routes, the functionality of the script is unaffected. 
#### How to verify it
run the script after adding a vxlan tunnel.

### CODE REVIEW

**Comment by prsunny** [comment]
We should not ignore vnet routes in the script. Scrip must handle the vnet routes in default scope and proceed with the check

**Comment by prsunny** [comment]
can you please add some tests and also fix the test failure?

File: scripts/vnet_route_check.py [hunk 1]
```diff
@@ -219,15 +219,24 @@ def get_vnet_routes_from_app_db():
             vnet_routes[vnet_name] = {}
             vnet_routes[vnet_name]['routes'] = []
 
-            intf = vnet_intfs[vnet_name][0]
-            vnet_routes[vnet_name]['vrf_oid'] = vnet_vrfs.get(intf, 'None')
+            if vnet_name not in vnet_intfs:
+                # this route has no vnet_intf and may be part of default VRF.
+                vnet_table = swsscommon.Table(db, 'VNET_TABLE')
+                scope = vnet_table.get(vnet_name)[1][1][1]
```
Comment (prsunny) [inline scripts/vnet_route_check.py:225]: what is this index [1][1][1] ? Can you provide an example?

File: scripts/vnet_route_check.py [hunk 2]
```diff
@@ -237,10 +246,12 @@ def get_vnet_routes_from_asic_db():
 
     vnet_vrfs = get_vrf_entries()
     vnet_vrfs_oids = [vnet_vrfs[k] for k in vnet_vrfs]
+    vnet_vrfs_oids.append(default_vrf_oid) 
 
     vnet_intfs = get_vnet_intfs()
 
     vrf_oid_to_vnet_map = {}
+    vrf_oid_to_vnet_map[default_vrf_oid] = 'default_VRF'
```
Comment (prsunny) [inline scripts/vnet_route_check.py:254]: I don't see this being used. Can you confirm?

File: scripts/vnet_route_check.py [hunk 3]
```diff
@@ -326,12 +350,17 @@ def main():
     # Don't run VNET routes consistancy logic if there is no VNET configuration
     if not check_vnet_cfg():
         return rc
+    asic_db = swsscommon.DBConnector('ASIC_DB', 0)
+    virtual_router = swsscommon.Table(asic_db, 'ASIC_STATE:SAI_OBJECT_TYPE_VIRTUAL_ROUTER')
+    default_vrf_oid = ""
+    if virtual_router.getKeys() != []:
+        default_vrf_oid = virtual_router.getKeys()[0]
```
Comment (prsunny) [inline scripts/vnet_route_check.py:357]: please make it as a global variable so you don't have to pass it in multiple places and use as "self.xxx"

File: scripts/vnet_route_check.py (hunk 1 above)
Comment (siqbal1986) [reply scripts/vnet_route_check.py:225]: this looking up the scope=default but i think I can improve it.

File: scripts/vnet_route_check.py (hunk 2 above)
Comment (siqbal1986) [reply scripts/vnet_route_check.py:254]: This is used as a placeholder, as the data structure keeps track of the routes by means of the vnet interface.

File: scripts/vnet_route_check.py (hunk 3 above)
Comment (siqbal1986) [reply scripts/vnet_route_check.py:357]: ill do so.

Comment (siqbal1986) [reply scripts/vnet_route_check.py:362]: done

**Review by prsunny — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**